test/unitTests/lspToolHost/configurationMiddleware.test.ts

```diff
@@ -45,6 +45,8 @@ const testData = [
     { serverOption: "csharp|code_style.formatting.indentation_and_spacing.indent_style", vsCodeConfiguration: "codeStyle.formatting.indentationAndSpacing.indentStyle", declareInPackageJson: false},
     { serverOption: "csharp|code_style.formatting.new_line.end_of_line", vsCodeConfiguration: "codeStyle.formatting.newLine.endOfLine", declareInPackageJson: false},
     { serverOption: "code_style.formatting.new_line.insert_final_newline", vsCodeConfiguration: "codeStyle.formatting.newLine.insertFinalNewline", declareInPackageJson: false},
+    { serverOption: "mystery_language|background_analysis.dotnet_solution_crawler_background_analysis_scope", vsCodeConfiguration: "dotnet.backgroundAnalysis.solutionCrawlerBackgroundAnalysisScope", declareInPackageJson: true},
```

Comment:
Each option should have two tests (because it is a per-lang option)
`csharp|background_analysis.dotnet_solution_crawler_background_analysis_scope` -> `dotnet.backgroundAnalysis.solutionCrawlerBackgroundAnalysisScope`
and
`mystery_language|background_analysis.dotnet_solution_crawler_background_analysis_scope` -> `null`

Only C# option should be marked as `declareInPackageJson: true`

It is in this way because for per-language option the server would ask the option value for all the supported languages.
And our client would only reply C# related options.

Reply:
Thanks @Cosifne - fixed now.